packages/docgen/lib/get-type-annotation.js

```diff
@@ -397,7 +397,10 @@ function getFunctionToken( token ) {
 
 function getFunctionNameForError( declarationToken ) {
```

Comment:
This particular function has been such a headache to maintain with its many edge cases. Thanks for taking the time to improve it!